NowPlaying.Core/Api/PlaybackStateProviderResolver.cs

```diff
@@ -0,0 +1,47 @@
+﻿using NowPlaying.Core.Api.Spotify;
+using NowPlaying.Core.Api.WindowsManager;
+using NowPlaying.Core.Settings;
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace NowPlaying.Core.Api
+{
+    public sealed class PlaybackStateProviderResolver
+    {
+        public Task<ITrackInfoUpdater> ResolveTrackInfoUpdater(PlaybackStateProvider provider)
+        {
+            switch (provider)
```

Comment:
https://docs.microsoft.com/en-us/archive/msdn-magazine/2019/may/csharp-8-0-pattern-matching-in-csharp-8-0#the-evolution-of-pattern-matching-in-c-80

```cs
return provider switch {
  ...
};
```

?